client/hosting/hosting-features/components/hosting-features.tsx

```diff
@@ -91,8 +94,11 @@ const HostingFeatures = () => {
 
 		if ( siteTransferData?.status === transferStates.COMPLETED ) {
 			dispatch( fetchAtomicTransfer( siteId ) as unknown as AnyAction );
+			queryClient.invalidateQueries( {
+				queryKey: queryKeyForAllSitesWithThemeSlug,
```

Comment:
We should use `getUseSiteExcerptsQueryKey` here. @nightnei and I discussed how we can ensure that we target the exact same query key that's used in `client/hosting/sites/components/sites-dashboard.tsx`, but I think we should consider **not doing that**. 

By specifying a shorter key when invalidating queries, `react-query` will invalidate all queries with longer keys matching that prefix. For example, if we invalidate the key `['sites']`, queries for `['sites', 'all']` will also be invalidated. 

This seems like an appropriate case to make use of that. We can invalidate all `useSiteExcerptsQuery` instances where `site_visibility === all` and just not specify `additional_fields` and `additional_options`.

Thoughts, @nightnei?